src/Warpstone/Internal/ParserImplementations/OptionalWhitespacesParserImpl.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections.Generic;
+using Warpstone.Internal.ParserExpressions;
+
+namespace Warpstone.Internal.ParserImplementations
+{
+    /// <summary>
+    /// Represents a parser that parses (optional) whitespaces.
+    /// </summary>
+    internal sealed class OptionalWhitespacesParserImpl : ParserImplementationBase<OptionalWhitespacesParser, string>
+    {
+        /// <summary>
+        /// Singleton instance of the parser.
+        /// </summary>
+        public static readonly OptionalWhitespacesParserImpl Instance = new();
```

Comment:
Add private default constructor to prevent other instances being created besides this singleton.